Abracadabra/AbracadabraAPI/AbracadabraAPI/Controllers/ApplicationsController.cs

```diff
@@ -60,6 +60,69 @@ public async Task<ActionResult<IEnumerable<ShortApplicationViewModel>>> GetShort
             return shortApplicationViewModels;
         }
 
+        //GET: api/Applications/all
+        [HttpGet("all")]
+        [Authorize(Roles = "Admin")]
+        public async Task<ActionResult<IEnumerable<ApplicationViewModel>>> GetApplications()
+        {
+            var user = await _userManager.FindByNameAsync(User.Identity.Name);
+
+            if (user == null)
+            {
+                return Unauthorized();
+            }
+
+            List<ExpertApplication> expertApplications = await _context.ExpertApplications.ToListAsync();
+            if (expertApplications.Count == 0)
+            {
+                return NoContent();
+            }
+
+            List<ApplicationViewModel> applicationViewModels = new List<ApplicationViewModel>();
+
+            foreach (var application in expertApplications)
+            {
+                if (application.Status == ApplicationStatus.Pending) {
+                    applicationViewModels.Add(Mapper.ApplicationToViewModel(application, await _context.Subjects.Where(x => x.ID == application.SubjectId).FirstOrDefaultAsync(), await  _userManager.FindByIdAsync(application.UserId)));
+                };
+            }
+
+
+            return applicationViewModels;
+        }
+
+        //GET: api/Applications/dashboard
+        [HttpGet("dashboard")]
+        [Authorize(Roles = "Admin")]
+        public async Task<ActionResult<IEnumerable<ApplicationViewModel>>> GetApplicationsForDashboard()
+        {
+            var user = await _userManager.FindByNameAsync(User.Identity.Name);
+
+            if (user == null)
+            {
+                return Unauthorized();
+            }
+
+            List<ExpertApplication> expertApplications = await _context.ExpertApplications.OrderBy(x => x.DateTimeCreated).ToListAsync();
```

Comment:
Will the OrderBy break if there are no expertApplications?

Reply:
No, in my understanding the order by orders all results it could get if that is null it wont order anything. And Just return the empty list